radicle-cob/src/object/collaboration/create.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2022 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use crate::Store;
+
+use super::*;
+
+/// The metadata required for creating a new [`CollaborativeObject`].
+pub struct Create<Author> {
+    /// The identity of the author for this object's first change.
+    pub author: Option<Author>,
+    /// The CRDT history to initialize this object with.
+    pub contents: Contents,
+    /// The typename for this object.
+    pub typename: TypeName,
+    /// An optional message to add when creating this object.
+    pub message: Option<String>,
+}
+
+impl<Author> Create<Author> {
+    fn create_spec(&self) -> change::Create<git_ext::Oid> {
+        change::Create {
+            typename: self.typename.clone(),
+            tips: Vec::new(),
+            message: self.message.clone(),
+            contents: self.contents.clone(),
+        }
+    }
+}
+
+/// Create a new [`CollaborativeObject`].
+///
+/// The `storage` is the backing storage for storing
+/// [`crate::Change`]s at content-addressable locations. Please see
+/// [`Store`] for further information.
+///
+/// The `signer` is expected to be a cryptographic signing key. This
+/// ensures that the objects origin is cryptographically verifiable.
+///
+/// The `resource` is the parent of this object, for example a
+/// software project.
+///
+/// The `args` are the metadata for this [`CollaborativeObject`]. See
+/// [`Create`] for further information.
+pub fn create<S, Signer, Person, Resource>(
+    storage: &S,
+    signer: Signer,
+    resource: &Resource,
+    args: Create<Person>,
```

Comment:
I'm seeing this "author" type sometimes be called "Person" and sometimes "Author"; I'd suggest "Author" is more applicable since it's not really related to human-ness. Perhaps this also relates to or resolves the question in git-commit around the "Person" type.

Reply:
Ya, I can change it :+1: